tests/test_external_search.py

```diff
@@ -54,259 +62,225 @@ def teardown(self):
+
+    def test_transfer_current_alias(self):
+        if not self.search:
+            return
+
+        # If the index doesn't exist, an error is raised.
+        assert_raises(
+            ValueError, self.search.transfer_current_alias, 'test_index-v3')
+
+        original_index = self.search.works_index
+
+        # If the -current alias doesn't exist, it's created
+        # and everything is updated accordingly.
+        self.search.indices.delete_alias(
+            index=original_index, name='test_index-current'
+        )
+        self.search.setup_index(new_index='test_index-v100')
+        self.search.transfer_current_alias('test_index-v100')
+        eq_('test_index-v100', self.search.works_index)
+        eq_('test_index-current', self.search.works_alias)
+
+        # If the -current alias already exists on the index,
+        # it's used without a problem.
+        self.search.transfer_current_alias('test_index-v100')
+        eq_('test_index-v100', self.search.works_index)
+        eq_('test_index-current', self.search.works_alias)
+
+        # If the -current alias is being used on a different version of the
+        # index, it's deleted from that index and placed on the new one.
+        self.search.setup_index(original_index)
+        self.search.transfer_current_alias(original_index)
+        eq_(original_index, self.search.works_index)
+        eq_('test_index-current', self.search.works_alias)
+
+        # It has been removed from other index.
+        eq_(False, self.search.indices.exists_alias(
+            index='test_index-v100', name='test_index-current'))
+        # And only exists on the new index.
+        alias_indices = self.search.indices.get_alias(name='test_index-current').keys()
+        eq_(['test_index-v0'], alias_indices)
+
+        # If the index doesn't have the same base name, an error is raised.
+        assert_raises(
+            ValueError, self.search.transfer_current_alias, 'banana-v10')
+
+class TestExternalSearchWithWorks(ExternalSearchTest):
+    """These tests run against a real search index with works in it.
+    The setup is very slow, so all the tests are in the same method.
+    Don't add new methods to this class - add more tests into test_query_works,
+    or add a new test class.
     """
 
     def setup(self):
-        super(TestExternalSearch, self).setup()
+        super(TestExternalSearchWithWorks, self).setup()
         if self.search:
-            works = []
 
             self.moby_dick = self._work(title="Moby Dick", authors="Herman Melville", fiction=True)
             self.moby_dick.presentation_edition.subtitle = "Or, the Whale"
             self.moby_dick.presentation_edition.series = "Classics"
             self.moby_dick.summary_text = "Ishmael"
             self.moby_dick.presentation_edition.publisher = "Project Gutenberg"
             self.moby_dick.set_presentation_ready()
-            works.append(self.moby_dick)
```

Comment:
I realized set_presentation_ready already adds everything to the search index, so the extra bulk update at the end of this method wasn't doing anything. It might be a little faster to create all the works before setting up the search index and do the bulk update, but if I create the search index first I can skip creating the works entirely if it doesn't work.